cassandra/pool.py

```diff
@@ -360,18 +363,41 @@ def __init__(self, host, host_distance, session):
             return
 
         log.debug("Initializing connection for host %s", self.host)
-        self._connection = session.cluster.connection_factory(host.endpoint)
+        self._connections[0] = session.cluster.connection_factory(host.endpoint)
```

Comment:
you got a point there, since we need to move all the logic of creating the mapping of connection to here (according to you other comment), I'll fix it when I move it here.